assets/assets/utils/chem_utils.py

```diff
@@ -0,0 +1,60 @@
+import os
+from typing import Union
+
+from indigo import Indigo
+from indigo.renderer import IndigoRenderer
+
+from assets import exceptions, logger
+
+SUPPORTED_FORMATS = {
+    ".pdf": "pdf",
+    ".mol": "chem",
+    ".sdf": "chem",
+    ".png": "image",
+    ".jpg": "image",
+    ".jpeg": "image",
+    ".txt": "text",
+}
+
+logger_ = logger.get_logger(__name__)
+
+MAX_MOLECULES_COUNT_IN_FILE = os.getenv("MAX_MOLECULES_COUNT_IN_FILE", 50)
+
+
+def get_page_count(file: bytes, ext: str) -> int:
+    indigo = Indigo()
+    pages: int = 1
+    if ext == ".sdf":
+        for pages, _ in enumerate(
+            indigo.iterateSDF(indigo.loadBuffer(file)), start=1
+        ):
```

Comment:
Maybe use range(len()) instead?

Reply:
Not sure I got your suggestion. Currently, we're iterating through sdf file until limit is not reached. How can we calculate count of items in sdf without reading it?